prototype_source/pt2e_quant_qat_static.rst

```diff
@@ -0,0 +1,470 @@
+graph mode based on `torch.export.export <https://pytorch.org/docs/main/export.html>`_.
+For more details about PyTorch 2.0 Export Quantization in general, refer
+to the `static post training quantization tutorial <https://pytorch.org/tutorials/prototype/pt2e_quant_ptq_static.html>`_.
+
+The PyTorch 2.0 Export QAT flow looks like the following. It is similar
+to the post training quantization (PTQ) flow for the most part:
+
+.. code:: python
+
+  import torch
+  from torch._export import capture_pre_autograd_graph
+  from torch.ao.quantization.quantize_pt2e import (
+    prepare_qat_pt2e,
+    convert_pt2e,
+  )
+  from torch.ao.quantization.quantizer import (
+    XNNPACKQuantizer,
+    get_symmetric_quantization_config,
+  )
+
+  class M(torch.nn.Module):
+     def __init__(self):
+        super().__init__()
+        self.linear = torch.nn.Linear(5, 10)
+
+     def forward(self, x):
+        return self.linear(x)
+
+
+  example_inputs = (torch.randn(1, 5),)
+  m = M()
+
+  # Step 1. program capture
+  # NOTE: this API will be updated to torch.export API in the future, but the captured
+  # result shoud mostly stay the same
+  m = capture_pre_autograd_graph(m, *example_inputs)
+  # we get a model with aten ops
+
+  # Step 2. quantization-aware training
+  # backend developer will write their own Quantizer and expose methods to allow
+  # users to express how they want the model to be quantized
+  quantizer = XNNPACKQuantizer().set_global(get_symmetric_quantization_config())
+  m = prepare_qat_pt2e(m, quantizer)
+
+  # train omitted
+
+  m = convert_pt2e(m)
+  # we have a model with aten ops doing integer computations when possible
+
+  # move the quantized model to eval mode, equivalent to `m.eval()`
+  torch.ao.quantization.move_exported_model_to_eval(m)
+
+Note that calling `model.eval()` or `model.train()` after program capture is
+not allowed, because these methods no longer correctly change the behavior of
+certain ops like dropout and batch normalization. Instead, please use
+`torch.ao.quantization.move_exported_model_to_eval()` and
+`torch.ao.quantization.move_exported_model_to_train()` (coming soon)
+respectively.
+
+
+Define Helper Functions and Prepare Dataset
+-------------------------------------------
+
+We’ll start by doing the necessary imports, defining some helper functions and
+prepare the data. These steps are very similar to the ones defined in the
+`static post training quantization tutorial <https://pytorch.org/tutorials/advanced/static_quantization_tutorial.html>`_.
+
+To run the code in this tutorial using the entire ImageNet dataset, first
+download Imagenet by following the instructions at here
+`ImageNet Data <http://www.image-net.org/download>`_. Unzip the downloaded file
+into the ``data_path`` folder.
+
+Download the `torchvision resnet18 model <https://download.pytorch.org/models/resnet18-f37072fd.pth>`_
+and rename it to ``data/resnet18_pretrained_float.pth``.
+
```

Comment:
You might want to add something like: In the following code block, we do blah... and describe what is happening there.